tpp-run/MLIRBench.cpp

```diff
@@ -238,10 +252,20 @@ LogicalResult MLIRBench::printMemRef(mlir::Value memRef) {
 
   // Vector undefined value
   APFloat vectorFloatValue = APFloat(-1.0F);
-  auto minusOne = builder.create<arith::ConstantFloatOp>(
-      unkLoc, vectorFloatValue, builder.getF32Type());
+  Value minusOne;
+  if (outputType.getElementType().isBF16()) {
+    bool Ignored;
+    vectorFloatValue.convert(APFloat::BFloat(), APFloat::rmNearestTiesToEven,
+                             &Ignored);
+
+    minusOne = builder.create<arith::ConstantFloatOp>(
+        unkLoc, vectorFloatValue, FloatType::getBF16(builder.getContext()));
+  } else {
+    minusOne = builder.create<arith::ConstantFloatOp>(unkLoc, vectorFloatValue,
+                                                      builder.getF32Type());
+  }
 
-  // Loop through memref, transfer each dim to vector
+  // Vector undefined value
```

Comment:
By "Vector undefined value" here I meant: When creating a vector, we can't leave `undef`, so we leave `-1.0` as the value that hasn't been filled by a memref read.

The following code is still looping through the memref...